model/search_pipeline/search_pipeline.smithy

```diff
@@ -0,0 +1,80 @@
+// SPDX-License-Identifier: Apache-2.0
+//
+//  The OpenSearch Contributors require contributions made to
+//  this file be licensed under the Apache-2.0 license or a
+//  compatible open source license.
+
+$version: "2"
+namespace OpenSearch
+
+structure SearchPipelineStructure{
+    version: Integer,
+    request_processors: RequestProcessorsList,
+    response_processors: ResponseProcessorsList
+}
+
+list RequestProcessorsList {
+    member: RequestProcessors
+}
+
+list ResponseProcessorsList {
+    member: RequestProcessors
+}
+
+
+structure RequestProcessors {
+    filter_query: FilterQuery,
+    neural_query_enricher: NeuralQueryEnricher,
+    script: SearchScript
+}
+
+structure ResponseProcessors {
+    personalize_search_ranking: PersonalizeSearchRanking,
+    rename_field: RenameField,
+}
```

Comment:
Really these are more like `union`s rather than `structure`s, but not sure how the OpenAPI conversion handles unions.

Reply:
Union will become `oneOf` in OpenAPI

Reply:
I never used unions. Do i need to remake this?